### ISSUE TITLE
logic error in tNMEA0183::Init(...)

### ISSUE THREAD

**Issue by lsoltero** [ttlappalainen/NMEA0183#22]
should this
bool tNMEA0183Msg::Init(const char *_MessageCode, const char *_Sender, char _Prefix) {
  Clear();
  size_t nSender=2;
  size_t nMessageCode=0;
  if ( _Sender==0 && (nSender=strlen(_Sender))>7 ) return false;

did you mean

  if ( _Sender==0 || (nSender=strlen(_Sender))>7 ) return false;

g++ is complaining...

**Comment by ttlappalainen** [comment]
Yes. Luckily _Sender has not been 0. I'll do update. There was an other fix too! I'll note you on version history - OK?
